### PR TITLE
Bigquery naming refactor

### PR DESCRIPTION
**Pull request by vdelacruzb**

### CODE REVIEW

File: common/bigquery/Makefile [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
 
+CLOUD = bigquery
+
 SED = sed
 GSUTIL = gsutil
 BQ = bq --location=$(BQ_REGION) --project_id=$(BQ_PROJECT)
 
+BQ_MODULE_LABEL ?= spatial_extension_module:core
+BQ_DEPLOY_DATASET = $(BQ_DATASET_PREFIX)carto
 BQ_DATASET = $(BQ_DATASET_PREFIX)$(MODULE)
```
Comment (jgoizueta) [inline common/bigquery/Makefile:11]: Could we rename this? (I assume it won't be used as a dataset name, right?) perhaps BQ_DEPLOY_MODULE ?

File: common/bigquery/Makefile [hunk 2]
```diff
@@ -102,30 +148,47 @@ storage-remove:
 	fi
 
 dataset-create:
-	$(BQ) show $(BQ_DATASET) 2>/dev/null 1>/dev/null || \
-		$(BQ) mk -d --description "$(BQ_DATASET) module" -label $(BQ_MODULE_CORE_LABEL) $(BQ_DATASET)
+	$(BQ) show $(BQ_DEPLOY_DATASET) 2>/dev/null 1>/dev/null || \
+		$(BQ) mk -d --description "$(BQ_DEPLOY_DATASET) module" -label $(BQ_MODULE_LABEL) $(BQ_DEPLOY_DATASET)
+ifdef BQ_CREATE_SPATIAL_CATALOG
```
Comment (jgoizueta) [inline common/bigquery/Makefile:153]: We could we replace this by a more generic mechanism, like a `BQ_POST_DATASET_CREATE` pointing to a shell script that we define in the data module, then we do something like this here:
```
ifdef BQ_POST_DATASET_CREATE
  BQ_POST_DATASET_CREATE
endif
```

File: common/bigquery/geonames/01_create_names.sql [hunk 3]
```diff
@@ -0,0 +1,118 @@
+CREATE OR REPLACE TABLE `@@BQ_PREFIX@@carto.names` (
```
Comment (jgoizueta) [inline common/bigquery/geonames/01_create_names.sql:1]: We'll remove this (it's handled by DO now)

File: common/bigquery/Makefile (hunk 1 above)
Comment (Jesus89) [reply common/bigquery/Makefile:11]: BQ_DEPLOY_DATASET is the dataset where the toolbox is installed.

Comment (Jesus89) [reply common/bigquery/Makefile:11]: I have changed `$(BQ_DATASET_PREFIX)$(MODULE)` to be used directly in the bucket variable

File: common/bigquery/Makefile (hunk 2 above)
Comment (Jesus89) [reply common/bigquery/Makefile:153]: Good point. I have checked that a strategy using a Makefile rule `post-dataset-create:` works

File: common/bigquery/geonames/01_create_names.sql (hunk 3 above)
Comment (Jesus89) [reply common/bigquery/geonames/01_create_names.sql:1]: :fire:

**Review by Jesus89 — APPROVED** [review]

**Pull request merged**